packages/teams-js/src/internal/messageObjects.ts

```diff
@@ -1,9 +1,55 @@
+import { generateGUID, validateUuid } from './utils';
+
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export class BaseUUID {
+  private uuid: string;
+
+  public constructor(uuid?: string) {
+    if (uuid) {
+      validateUuid(uuid);
+      this.uuid = uuid;
+    } else {
+      this.uuid = generateGUID();
+    }
+  }
+
+  public getUuidValue(): string {
+    return this.uuid;
+  }
+}
+
+export class MessageUUID extends BaseUUID {}
```

Comment:
out of curiosity, what does having this extend BaseUUID get you? It seems like you could delete this line and rename BaseUUID to MessageUUID and you would be at basically the same place.

Reply:
Maybe you are thinking there would be ones that are for things other than Message, although that still might just warrant renaming BaseUUID to something like UUID.

Reply:
Ya I was thinking about this being extensible down the line, but might as well address that in the future instead of trying to make an empty object that handles it now.